### PR TITLE
Add device registry for MQTT sensors

### PR DESCRIPTION
**Pull request by OttoWinter**
**Description:**

Add device registry integration to MQTT sensors. The plan is to expand it to other platforms too.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#16943

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

**Review by DubhAd — DISMISSED** [review]
👍

File: source/_components/sensor.mqtt.markdown [hunk 1]
```diff
@@ -91,6 +91,35 @@ device_class:
   required: false
   type: device_class
   default: None
+device:
+  description: 'Information about the device this sensor is a part of to tie it into the [device registry](https://developers.home-assistant.io/docs/en/device_registry_index.html). Only works through [MQTT discovery](/docs/mqtt/discovery/) and when [`unique_id`](#unique_id) is set.'
```
Comment (frenck) [inline source/_components/sensor.mqtt.markdown:95]: Some commas would make it more readable (just like the other PRs).

File: source/_components/sensor.mqtt.markdown [hunk 2]
```diff
@@ -91,6 +91,35 @@ device_class:
...
+  type: map
+  keys:
+    identifiers:
+      description: 'A list of IDs that uniquely identify the device. For example a serial number.'
+      required: false
+      type: list, string
+    connections:
+      description: 'A list of connections of the device to the outside world as a list of tuples `[connection_type, connection_identifier]`. For example the MAC address of a network interface: `"connections": [["mac", "02:5b:26:a8:dc:12"]]`.'
+      required: false
+      type: list, tuple
```
Comment (frenck) [inline source/_components/sensor.mqtt.markdown]: `tuple` is not a valid documentation configuration type.

**Review by frenck — CHANGES_REQUESTED** [review]
@OttoWinter Left you some comments, could you please take a look?

**Review by frenck — APPROVED** [review]

**Pull request merged**